velox/common/base/Counters.cpp

```diff
@@ -19,47 +19,51 @@
 
 namespace facebook::velox {
 
-void registerVeloxCounters() {
+void registerVeloxMetrics() {
   // Tracks hive handle generation latency in range of [0, 100s] and reports
   // P50, P90, P99, and P100.
-  REPORT_ADD_HISTOGRAM_EXPORT_PERCENTILE(
-      kCounterHiveFileHandleGenerateLatencyMs, 10, 0, 100000, 50, 90, 99, 100);
+  DEFINE_HISTOGRAM_METRIC(
+      kMetricHiveFileHandleGenerateLatencyMs, 10, 0, 100000, 50, 90, 99, 100);
 
-  REPORT_ADD_STAT_EXPORT_TYPE(
-      kCounterCacheShrinkCount, facebook::velox::StatType::COUNT);
+  DEFINE_METRIC(kMetricCacheShrinkCount, facebook::velox::StatType::COUNT);
 
   // Tracks cache shrink latency in range of [0, 100s] and reports P50, P90,
   // P99, and P100.
-  REPORT_ADD_HISTOGRAM_EXPORT_PERCENTILE(
-      kCounterCacheShrinkTimeMs, 10, 0, 100'000, 50, 90, 99, 100);
+  DEFINE_HISTOGRAM_METRIC(
+      kMetricCacheShrinkTimeMs, 10, 0, 100'000, 50, 90, 99, 100);
 
   // Tracks memory reclaim exec time in range of [0, 600s] and reports
   // P50, P90, P99, and P100.
-  REPORT_ADD_HISTOGRAM_EXPORT_PERCENTILE(
-      kCounterMemoryReclaimExecTimeMs, 20, 0, 600'000, 50, 90, 99, 100);
+  DEFINE_HISTOGRAM_METRIC(
+      kMetricMemoryReclaimExecTimeMs, 20, 0, 600'000, 50, 90, 99, 100);
 
   // Tracks memory reclaim task wait time in range of [0, 60s] and reports
   // P50, P90, P99, and P100.
-  REPORT_ADD_HISTOGRAM_EXPORT_PERCENTILE(
-      kCounterMemoryReclaimWaitTimeMs, 10, 0, 60'000, 50, 90, 99, 100);
+  DEFINE_HISTOGRAM_METRIC(
+      kMetricMemoryReclaimWaitTimeMs, 10, 0, 60'000, 50, 90, 99, 100);
 
   // Tracks memory reclaim bytes.
-  REPORT_ADD_STAT_EXPORT_TYPE(
-      kCounterMemoryReclaimedBytes, facebook::velox::StatType::SUM);
+  DEFINE_METRIC(kMetricMemoryReclaimedBytes, facebook::velox::StatType::SUM);
 
   // Tracks the number of times that the memory reclaim wait timeouts.
-  REPORT_ADD_STAT_EXPORT_TYPE(
-      kCounterMemoryReclaimWaitTimeoutCount, facebook::velox::StatType::SUM);
+  DEFINE_METRIC(
+      kMetricMemoryReclaimWaitTimeoutCount, facebook::velox::StatType::SUM);
 
   // Tracks the number of times that the memory reclaim fails because of
   // non-reclaimable section which is an indicator that the memory reservation
   // is not sufficient.
-  REPORT_ADD_STAT_EXPORT_TYPE(
-      kCounterMemoryNonReclaimableCount, facebook::velox::StatType::COUNT);
+  DEFINE_METRIC(
+      kMetricMemoryNonReclaimableCount, facebook::velox::StatType::COUNT);
 
   // Tracks the number of times that we hit the max spill level limit.
-  REPORT_ADD_STAT_EXPORT_TYPE(
-      kCounterMaxSpillLevelExceededCount, facebook::velox::StatType::COUNT);
+  DEFINE_METRIC(
+      kMetricMaxSpillLevelExceededCount, facebook::velox::StatType::COUNT);
+}
+
+#ifdef VELOX_ENABLE_BACKWARD_COMPATIBILITY
```

Comment:
This macro doesn't work in .cpp file. You need to move the implementation of registerVeloxCounters into .h file.